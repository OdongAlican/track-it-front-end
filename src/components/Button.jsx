import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

const Button = ({ detail }) => (
  <NavLink
    className="nav-link"
    style={
            {
              color: '#ABAEB3',
              textDecoration: 'none',
              padding: 0,
            }
        }
    to={{
      pathname: `${detail.path}`,
    }}
    activeClassName="active-button"
  >
    <div data-testid="appButton" className="link-div">
      <i className={detail.className} />
      <p>{ detail.name }</p>
    </div>

  </NavLink>
);

Button.propTypes = {
  detail: PropTypes.shape({
    path: PropTypes.string,
    className: PropTypes.string,
    name: PropTypes.string,
  }).isRequired,
};

export default Button;
