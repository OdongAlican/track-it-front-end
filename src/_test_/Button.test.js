import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Button from '../components/Button';

afterEach(cleanup);
const initialState = {
  authReducer: {
    output: 10,
    error: 'test',
  },
};

const detail = {
  path: '/test-path',
  className: 'sample color',
  name: 'test name',
};

const mockStore = configureStore();
let store;

it('renders Button correctly', () => {
  store = mockStore(initialState);
  const { getAllByTestId } = render(
    <Router>
      <Provider store={store}><Button detail={detail} /></Provider>
    </Router>,
  );
  const result = getAllByTestId('appButton');
  expect(result[0]).toHaveTextContent('test name');
});

it('renders Button correctly', () => {
  store = mockStore(initialState);
  const { getAllByTestId } = render(
    <Router>
      <Provider store={store}><Button detail={detail} /></Provider>
    </Router>,
  );
  const result = getAllByTestId('appButton');
  expect(result[0]).not.toHaveTextContent('some random text');
});

it('matches Button snapshot', () => {
  const appointment = renderer.create(
    <Router>
      <Provider store={store}><Button detail={detail} /></Provider>
      ,
    </Router>,
  );
  const tree = appointment.toJSON();
  expect(tree).toMatchSnapshot();
});
