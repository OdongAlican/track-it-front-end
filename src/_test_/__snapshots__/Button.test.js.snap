// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Button snapshot 1`] = `
Array [
  <div
    className="link-div px-0"
    data-testid="appButton"
  >
    <button
      className="link-button"
      type="button"
    >
      <a
        href="/test-path"
        onClick={[Function]}
        style={
          Object {
            "color": "#ABAEB3",
          }
        }
      >
        <i
          className="sample color"
        />
        <p>
          test name
        </p>
      </a>
    </button>
  </div>,
  ",",
]
`;
